service/src/main/java/greencity/service/ubs/UBSClientServiceImpl.java

```diff
@@ -877,125 +829,25 @@ private void setLocations(CreateAddressRequestDto addressRequestDto, Address add
-        GeocodingResult resultsEn = googleApiService.getResultFromGeoCode(placeId, 1);
-
-        OrderAddressDtoRequest orderAddressDtoRequest = new OrderAddressDtoRequest();
-        initializeGeoCodingResults(initializeUkrainianGeoCodingResult(orderAddressDtoRequest), resultsUa);
-        initializeGeoCodingResults(initializeEnglishGeoCodingResult(orderAddressDtoRequest), resultsEn);
-
-        double latitude = resultsEn.geometry.location.lat;
-        double longitude = resultsEn.geometry.location.lng;
-        orderAddressDtoRequest.setCoordinates(new Coordinates(latitude, longitude));
-
-        checkIfAddressBelongToKyiv(orderAddressDtoRequest);
-
-        return orderAddressDtoRequest;
-    }
-
-    /**
-     * When Google API sets region name, there's a special case for Kyiv, the
-     * capital of Ukraine. Google API sets the 'Kyiv' as the name of region for
-     * addresses from Kyiv instead of Kyiv Region. Therefore, a separate logic is
-     * implemented to set 'Kyiv Region' name for such addresses instead of 'Kyiv'.
-     *
-     * @param request OrderAddressDtoRequest.
-     */
-    private void checkIfAddressBelongToKyiv(OrderAddressDtoRequest request) {
-        if (request.getRegion().equalsIgnoreCase(KYIV_UA)) {
-            request.setRegion(KYIV_REGION_UA);
-        }
-
-        if (request.getRegionEn().equalsIgnoreCase(KYIV_EN)) {
-            request.setRegionEn(KYIV_REGION_EN);
-        }
-    }
-
-    private void checkIfAddressBelongToKyiv(Region region, Address address) {
-        if (region.getNameUk().equalsIgnoreCase(KYIV_UA)) {
-            address.setRegion(KYIV_REGION_UA);
-            // todo: set Kyiv oblast region id
-        }
-
-        if (region.getNameEn().equalsIgnoreCase(KYIV_EN)) {
-            address.setRegionEn(KYIV_REGION_EN);
-            // todo: set Kyiv oblast region id
-        }
-    }
-
-    private void checkNullFieldsOnGoogleResponse(OrderAddressDtoRequest dtoRequest,
-        OrderAddressDtoRequest addressRequestDto) {
-        dtoRequest.setRegion(
-            Objects.isNull(dtoRequest.getRegion()) ? addressRequestDto.getRegion() : dtoRequest.getRegion());
-
-        dtoRequest.setRegionEn(
-            Objects.isNull(dtoRequest.getRegionEn()) ? addressRequestDto.getRegionEn() : dtoRequest.getRegionEn());
-
-        dtoRequest.setDistrict(
-            Objects.isNull(dtoRequest.getDistrict()) ? addressRequestDto.getDistrict() : dtoRequest.getDistrict());
-
-        dtoRequest.setDistrictEn(
-            Objects.isNull(dtoRequest.getDistrictEn()) ? addressRequestDto.getDistrictEn()
-                : dtoRequest.getDistrictEn());
 
-        dtoRequest.setHouseNumber(
-            Objects.isNull(dtoRequest.getHouseNumber()) ? addressRequestDto.getHouseNumber()
-                : dtoRequest.getHouseNumber());
+        Optional<Address> deletedAddress = addresses.stream()
+            .filter(address -> address.getAddressStatus().equals(AddressStatus.DELETED))
```

Comment:
please use AddressStatus.DELETED. equals(address.getAddressStatus()) to prevent NPE

Reply:
done